src/crimson/os/seastore/segment_cleaner.h

```diff
@@ -272,10 +272,7 @@ class segment_info_set_t {
  */
 class SegmentProvider {
 public:
-  using get_segment_ertr = crimson::errorator<
-    crimson::ct_error::input_output_error>;
-  using get_segment_ret = get_segment_ertr::future<segment_id_t>;
-  virtual get_segment_ret get_segment(
+  virtual segment_id_t get_segment(
```

Comment:
Seems we cannot really block here -- seastore cannot release segments if cleaner transactions are blocked here due to out-of-space.

I think we already have `reserve_projected_usage()` to block user transactions and make sure there is always enough space for cleaning.